ci/test/04_install.sh

```diff
@@ -78,9 +72,16 @@ elif [ "$CI_USE_APT_INSTALL" != "no" ]; then
   fi
   ${CI_RETRY_EXE} CI_EXEC apt-get update
   ${CI_RETRY_EXE} CI_EXEC apt-get install --no-install-recommends --no-upgrade -y "$PACKAGES" "$DOCKER_PACKAGES"
-  if [ -n "$PIP_PACKAGES" ]; then
+fi
+
+if [ -n "$PIP_PACKAGES" ]; then
+  if [ "$CI_OS_NAME" == "macos" ]; then
+    sudo -H pip3 install --upgrade pip
+    # shellcheck disable=SC2086
+    IN_GETOPT_BIN="/usr/local/opt/gnu-getopt/bin/getopt" ${CI_RETRY_EXE} pip3 install --user $PIP_PACKAGES
+  else
     # shellcheck disable=SC2086
-    ${CI_RETRY_EXE} pip3 install --user $PIP_PACKAGES
+    ${CI_RETRY_EXE} CI_EXEC pip3 install --user $PIP_PACKAGES
```

Comment:
Review note: Adding CI_EXEC is required, otherwise it would result in a failure.

To reproduce the failure:


```
$ RESTART_CI_DOCKER_BEFORE_RUN=1 FILE_ENV="./ci/test/00_setup_env_i686_centos.sh" ./ci/test_run_all.sh
...
bash: line 1: pip3: command not found

Reply:
This also fixes a bug on historic commits. To reproduce the bug on the historic commits:

```
$ git checkout 3c3bd9022026a75e15492ba9cf8bf3b72866b9a9~ && RESTART_CI_DOCKER_BEFORE_RUN=1 FILE_ENV="./ci/test/00_setup_env_i686_multiprocess.sh" ./ci/test_run_all.sh
...
bash: line 1: pip3: command not found